src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -733,9 +733,9 @@ ValueTask<string> GetUsernameAsync(bool async, CancellationToken cancellationTok
 
         async ValueTask<string> GetUsernameAsyncInternal()
         {
-            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && DataSource.IntegratedSecurityHandler.IsSupported)
             {
-                username = await KerberosUsernameProvider.GetUsernameAsync(Settings.IncludeRealm, ConnectionLogger, async,
+                username = await DataSource.IntegratedSecurityHandler.GetUsername(async, Settings.IncludeRealm, ConnectionLogger,
```

Comment:
This is the only place where we call `IntegratedSecurityHandler.GetUsername`. So by checking `IntegratedSecurityHandler.IsSupported` default `IntegratedSecurityHandler.GetUsername` implementation is pretty much pointless (since it will never throw an exception).
What if instead we always call `IntegratedSecurityHandler.GetUsername` but the default implementation just logs that integrated security is disabled (same thing as `KerberosUsernameProvider.GetUsername` does)?

Reply:
Good point, changed.